File: appmonet-sdk/appmonet-sdk-mopub/src/main/java/com/monet/bidder/MoPubAdServerWrapper.java
package com.monet.bidder;

import com.monet.bidder.auction.AuctionRequest;

class MoPubAdServerWrapper implements AdServerWrapper {

  @Override
  public AdServerAdRequest newAdRequest(AuctionRequest auctionRequest) {
    return null;
  }

  @Override
  public AdServerAdRequest newAdRequest(AuctionRequest auctionRequest, Type type) {
    if (type == Type.INTERSTITIAL) {
      return MoPubInterstitialAdRequest.fromAuctionRequest(auctionRequest);
    } else if (type == Type.NATIVE) {
      return MopubNativeAdRequest.fromAuctionRequest(auctionRequest);
    } else {
      return MoPubAdRequest.fromAuctionRequest(auctionRequest);
    }
  }

  @Override
  public AdServerAdRequest newAdRequest() {
    return new MoPubAdRequest();
  }

  @Override
  public AdSize newAdSize(Integer width, Integer height) {
    return new AdSize(width, height);
  }
}

File: appmonet-sdk/appmonet-sdk-mopub/src/main/java/com/monet/bidder/MoPubAdServerWrapper.kt
package com.monet.bidder

import android.content.Context
import com.monet.AdServerAdRequest
import com.monet.adview.AdSize
import com.monet.AdServerWrapper
import com.monet.AdType
import com.monet.AdType.INTERSTITIAL
import com.monet.AdType.NATIVE
import com.monet.auction.AuctionRequest

internal class MoPubAdServerWrapper(private val context: Context) : AdServerWrapper {
  override fun newAdRequest(
    auctionRequest: AuctionRequest,
    type: AdType
  ): AdServerAdRequest {
    return when (type) {
      INTERSTITIAL -> {
        MoPubInterstitialAdRequest.fromAuctionRequest(auctionRequest)
      }
      NATIVE -> {
        MopubNativeAdRequest.fromAuctionRequest(auctionRequest)
      }
      else -> {
        MoPubAdRequest.fromAuctionRequest(auctionRequest)
      }
    }
  }

  override fun newAdRequest(): AdServerAdRequest {
    return MoPubAdRequest()
  }

  override fun newAdSize(
    width: Int,
    height: Int
  ): AdSize {
    return AdSize(context, width, height)
  }
}